03-Binary-Search-Trees/01-Binary-Search-Trees.md

```diff
@@ -63,59 +82,55 @@ Our Linked Lists are a linear structure with each node linking to the next node
 
 ![Linked List Diagram](images/linked-list-vocab.png)
 
-Our ListNode looked like this:
-
-```ruby
-class Node
-  attr_reader :data
-  attr_accessor :next
+Our Node class from the LinkedList topic, which will be referred to as `ListNode`, looked like this:
 
-  def initialize(value, next = nil)
-    @next = next
-    @data = value
-  end
-end
+```python
+class ListNode:
+    def __init__(self, value, next=None):
+        self.data = value
+        self.next = next
 ```
 
-The `ListNode` class was used in a larger `LinkedList` class which maintained a chain of `ListNode` objects starting with a node pointed to by an instance variable called `@head`.
+The `ListNode` class was used in a larger `LinkedList` class which maintained a chain of `ListNode` objects starting with a node pointed to by an instance variable called `head`.
 
 ### Consider A Nonlinear Structure...
 
 In a _Binary Search Tree_ each node's left pointer points to all elements smaller than or equal to the node's key.  The right pointer points to all nodes greater than the given node's key.   Each node can refer to other nodes.  A Tree is hierarchical with certain nodes acting as parents to others.  A node above another is the node's _parent_.  The node(s) below a node are it's _children_.  The topmost node in a tree is known as the _root_.  The nodes with no children are called _leaves_.
 
-![Binary Search Tree Vocabular](images/TreeVocabulary.png)
+In a Binary Search Tree:
+- Nodes with values less than any node are stored to the **left** of that node.
+- Nodes with values greater than any node are stored to the **right** of that node. 
 
+![Binary Search Tree Vocabular](images/TreeVocabulary.png)
 
 ### Binary Search Tree Node
 
 Instead of a node with one link to a next node, we can create nodes with 2 pointers, left and right.  Since each node can have 2 successors or "children", it forms a binary structure as opposed to the linear structure of a Linked List.
 
-```ruby
-class TreeNode
-  attr_reader :key, :data
-  attr_accessor :left, :right
+This node stores both a key and value for each node.  The `Tree` class will compare keys to maintain node order.
 
-  def initialize(data)
-    @data = data
-    @left = @right = nil
-  end
-end
+```python
+class TreeNode:
+    def __init__(self, key, val = None):
+        if val == None:
+            val = key
+
+        self.key = key
+        self.value = val
+        self.left = None
+        self.right = None```
```

Comment:
The diff makes it look like you duplicated ` ``` `s:

```suggestion
        self.right = None
```